aoc/d3/main.go

```diff
@@ -0,0 +1,72 @@
+package main
+
+import (
+	"fmt"
+	"os"
+	"strings"
+)
+
+// split a string in two using the middle point
+func SplitString(input string) []string {
+	half := len(input) / 2
+	return []string{input[:half], input[half:]}
+}
+
+// find the character that exists in two strings
+func FindCommonItem(input1 string, input2 string) string {
+	var result string
+	for _, item := range input1 { // generated by ChatGPT
```

Comment:
This was purely for fun. It was a simple enough function that I wanted to see if it could be generated for me :D I didn't think it was great but it worked, and it seems you didn't like it either

Reply:
Yeah I mean it works but go being a relatively newish language probably didn't have enough sample data to provide good, idiomatic code. Copilot probably does better but even then I've found it useful only for trivial things like auxiliary functions/modules (even in Python).basically if it's on GitHub it's probably a library so I shouldn't be rewriting that code and if it's not one, it's either an auxiliary function or not exactly what I need 😅 and many of these functions I might just be able to write quicker than writing the comment copilot needs 😂